unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/BuilderMode/Controllers/BIWCreatorController.cs

```diff
@@ -172,10 +171,11 @@ public DCLBuilderInWorldEntity CreateCatalogItem(CatalogItem catalogItem, Vector
 
     public bool ExistsLoadingGameObjectForEntity(string entityId) { return loadingGameObjects.ContainsKey(entityId); }
 
-    private void CreateLoadingObject(DCLBuilderInWorldEntity entity)
+    public void CreateLoadingObject(DCLBuilderInWorldEntity entity)
     {
         BIWLoadingPlaceHolder loadingPlaceHolder = GameObject.Instantiate(loadingObjectPrefab, entity.gameObject.transform).GetComponent<BIWLoadingPlaceHolder>();
         loadingGameObjects.Add(entity.rootEntity.entityId, loadingPlaceHolder);
+        entity.OnShapeFinishLoading += OnShapeLoadFinish;
```

Comment:
Should we unsubscribe this in some place?

Reply:
We unsuscribe in the line 186 in the Function `OnShapeLoadFinish` this function is called even if the shape fails to load so we are sure that we are unsubscribing!